From: Liang Yan <lyan@suse.com>
Date: Fri, 12 Jul 2019 09:33:19 -0400
Subject: [PATCH] kvm/mmu: kABI fix for *_mmu_pages changes in struct kvm_arch
Patch-mainline: Never, kABI fix
References: bsc#1135335

Hide the kabi break for type changes of *_mmu_pages in struct kvm_arch. 
Related patch: 
patches.fixes/kvm-mmu-Fix-overflow-on-kvm-mmu-page-limit-calculati.patch

Signed-off-by: Liang Yan <lyan@suse.com>
---
 arch/x86/include/asm/kvm_host.h |   12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -778,9 +778,15 @@ struct kvm_sev_info {
 };
 
 struct kvm_arch {
-	unsigned long n_used_mmu_pages;
-	unsigned long n_requested_mmu_pages;
-	unsigned long n_max_mmu_pages;
+#ifndef __GENKSYMS__
+        unsigned long n_used_mmu_pages;
+        unsigned long n_requested_mmu_pages;
+        unsigned long n_max_mmu_pages;
+#else
+        unsigned int n_used_mmu_pages;
+        unsigned int n_requested_mmu_pages;
+        unsigned int n_max_mmu_pages;
+#endif
 	unsigned int indirect_shadow_pages;
 	unsigned long mmu_valid_gen;
 	struct hlist_head mmu_page_hash[KVM_NUM_MMU_PAGES];
